community/kernel/src/test/java/org/neo4j/kernel/impl/proc/ResourceInjectionTest.java

```diff
@@ -71,11 +74,22 @@ public void shouldFailNicelyWhenUnknownAPI() throws Throwable
         List<CallableProcedure> procList = compile( procedureWithUnknownAPI.class, true );
 
         //Then
-        verify( log ).warn(  "Unable to set up injection for procedure `procedureWithUnknownAPI`," +
-                " the field `api` has type `class org.neo4j.kernel.impl.proc.ResourceInjectionTest$UnknownAPI`" +
-                " which is not a known injectable component.");
+        verify( log )
+                .warn( "Unable to set up injection for procedure `procedureWithUnknownAPI`, the field `api` has type" +
+                        " `class org.neo4j.kernel.impl.proc.ResourceInjectionTest$UnknownAPI`" +
+                        " which is not a known injectable component." );
 
-        assert(procList.size() == 0);
+        assertThat( procList.size(), equalTo( 1 ) );
+        try
+        {
+            procList.get( 0 ).apply( new BasicContext(), new Object[0] );
+            fail();
+        }
+        catch ( ProcedureException e )
+        {
+            assertThat( e.getMessage(), containsString(
+                    "org.neo4j.kernel.impl.proc.listCoolPeople is not available due to not having access to one or more components." ) );
```

Comment:
As mentioned in a comment on an earlier PR, I would like to get separate messages for the cases 1) where the component exists but the procedure lacks access, and 2) when the component simply does not exist.